website/public/js/controllers/inventoryCtrl.js

```diff
@@ -103,6 +103,18 @@ habitrpg.controller("InventoryCtrl",
       }
     }
 
+    $scope.sellAllInventory = function() {
+      var selected = $scope.selectedEgg ? 'selectedEgg' : $scope.selectedPotion ? 'selectedPotion' : $scope.selectedFood ? 'selectedFood' : undefined;
+      if (selected) {
+        var type = $scope.selectedEgg ? 'eggs' : $scope.selectedPotion ? 'hatchingPotions' : $scope.selectedFood ? 'food' : undefined;
+        var amount = user.items[type][$scope[selected].key]; // user.items changes with every iteration of sell, so need to store in a variable first
+        for (i = 0; i < amount; i++) {
+          user.ops.sell({params:{type:type, key: $scope[selected].key}});
```

Comment:
Rather than calling a for loop, I think it would be better to adjust the user ops function to accept a quantity and have it default to 1.